apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -54,29 +66,205 @@ const TutorialExplorer = React.createClass({
       const itemIndex = this.state.filters[filterGroup].indexOf(filterEntry);
       newState = state.updateIn(['filters', filterGroup], arr => arr.splice(itemIndex, 1));
     }
-    this.setState(newState.toJS());
+
+    this.setState(newState.toJS(), function () {
+      const { filteredTutorials, filteredTutorialsForLocale } = this.filterTutorials(this.state.filters);
+      this.setState({
+        filteredTutorials: filteredTutorials,
+        filteredTutorialsCount: filteredTutorials.length,
+        filteredTutorialsForLocale: filteredTutorialsForLocale
+      });
+    });
```

Comment:
TIL - That you can pass setState a callback.

That said, I don't understand why we need to do so here. Couldnt we do something like this:
```
newState = newState.toJS();
const { filteredTutorials, filteredTutorialsForLocale } = this.filterTutorials(newState.filters);
this.setState({
   ...newState,
   filteredTutorials,
   filteredTutorialsForLocale
});